_docs-sources/pipelines/tutorial/index.md

```diff
@@ -1,3 +1,159 @@
+
+First, create the file structure that will contain the infrastructure unit that defines an S3 bucket in your environment. Replace `<account name>` with the value you used for `AwsAccountName` when generating your `infrastructure-live` code in the [infrastructure live](#infrastructure-live) section.
+
+```bash
+mkdir -p <account name>/us-east-1/<account name>/data-storage/s3
+touch <account name>/us-east-1/<account name>/data-storage/s3/terragrunt.hcl
+```
+
+Next, add the terragrunt code to create an S3 bucket. Copy the terragrunt code below, replacing `<your S3 bucket name>` with your desired bucket name. S3 bucket names need to be globally unique, so we've provided a helper script below to help generate the name of your bucket. You may name the bucket whatever you like, just make sure it’s unique.
+
+```bash
+export export UNIQUE_ID=$(uuidgen | tr 'A-Z' 'a-z')
+export DATE_NOW=$(date "+%F")
+echo "gwp-bucket-${UNIQUE_ID}-${DATE_NOW}"
+```
+
+```hcl title="<account name>/us-east-1/<account name>/data-storage/s3/terragrunt.hcl"
+# ------------------------------------------------------------------------------------------------------
+# DEPLOY GRUNTWORK’s S3-BUCKET MODULE
+# ------------------------------------------------------------------------------------------------------
+
+terraform {
+  source = "git::git@github.com:gruntwork-io/terraform-aws-service-catalog.git//modules/data-stores/s3-bucket?ref=v0.104.15"
+}
+
+include "root" {
+  path = find_in_parent_folders()
+}
+
+inputs = {
+  primary_bucket = "<your S3 bucket name>"
+}
+```
+
+### Planning and Applying
+
+Create a new branch for your changes, commit your changes to your branch, then push your branch. Next, create a PR to merge your branch into `main` (the default branch in your repository). Follow this GitHub tutorial to [create a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) using your preferred tool of choice.
+
+After you create the PR, GitHub Actions will automatically run the workflow defined in `pipelines.yml`. Once complete, pipelines will add a comment to the PR with a link to the workflow logs, click the link see the output of the `terragrunt plan` action that ran as a result of your changes.
+
+If you the `plan` output looks as expected, you are ready to merge your PR and create the S3 bucket. Click the `Merge pull request` button to merge the pull request. On merge, pipelines will run an `apply` action to provision the S3 bucket. You can find the Workflow run associated with the merged PR by navigating to the `main` branch on your PR and clicking on the Check Run icon at the top of the file explorer, then clicking `details` next to the pipelines workflow. This will take you to the `dispatch` job for pipelines, which contains a link to the workflow run in `infrastructure-pipelines` where you can see the output of your pipelines run.
```

Comment:
```suggestion
If the `plan` output looks as expected, you are ready to merge your PR and create the S3 bucket. Click the `Merge pull request` button to merge the pull request. On merge, pipelines will run an `apply` action to provision the S3 bucket. You can find the Workflow run associated with the merged PR by navigating to the `main` branch on your PR and clicking on the Check Run icon at the top of the file explorer, then clicking `details` next to the pipelines workflow. This will take you to the `dispatch` job for pipelines, which contains a link to the workflow run in `infrastructure-pipelines` where you can see the output of your pipelines run.
```